Tchap/Modules/Favourites/FavouriteMessages/FavouriteMessagesViewController.swift

```diff
@@ -255,19 +305,142 @@ extension FavouriteMessagesViewController: UITableViewDelegate {
-        return defaultValue;
+        var shouldDoAction = defaultValue
+        
+        guard let favouriteMessagesCell = cell as? MXKRoomBubbleTableViewCell else {
+            fatalError("MXKRoomBubbleTableViewCell is not of the expected class")
+        }
+        
+        guard let cellData = favouriteMessagesCell.bubbleData as? FavouriteMessagesBubbleCellData else {
+            fatalError("FavouriteMessagesBubbleCellData is not of the expected class")
+        }
+        
+        // Try to catch universal link supported by the app
+        if actionIdentifier == kMXKRoomBubbleCellShouldInteractWithURL, let url = userInfo[kMXKRoomBubbleCellUrl] as? URL {
+            
+            // When a link refers to a room alias/id, a user id or an event id, the non-ASCII characters (like '#' in room alias) has been escaped
+            // to be able to convert it into a legal URL string.
+            guard let absoluteURLString = url.absoluteString.removingPercentEncoding else {
+                fatalError("absoluteURLString should be defined")
+            }
+            
+            // Check whether this is a permalink to handle it directly into the app
+            if Tools.isPermaLink(url) {
+                // Patch: catch up all the permalinks even if they are not all supported by Tchap for the moment,
+                // like the permalinks with a userid.
+                shouldDoAction = false
+                
+                // iOS Patch: fix urls before using it
+                let fixedURL = Tools.fixURL(withSeveralHashKeys: url)
+                // In some cases (for example when the url has multiple '#'), the '%' character has been espaced twice in the provided url (we got %2524 for '$').
+                // We decided to remove percent encoding on all the fragment here. A second attempt will take place during the parameters parsing.
+                if let fragment = fixedURL?.fragment?.removingPercentEncoding {
+                    self.viewModel.process(viewAction: .tapAction(fragment: fragment))
+                }
+            
+            // Click on a member. Do nothing in favourites case.
+            } else if MXTools.isMatrixUserIdentifier(absoluteURLString) {
+                    print("[FavouriteMessagesViewController] showMemberDetails: Do nothing: \(absoluteURLString)")
+                
+            // Open the clicked room
+            } else if MXTools.isMatrixRoomIdentifier(absoluteURLString) || MXTools.isMatrixRoomAlias(absoluteURLString) {
+                shouldDoAction = false
+                self.viewModel.process(viewAction: .tapAction(fragment: absoluteURLString))
+            } else if absoluteURLString.hasPrefix(EventFormatterOnReRequestKeysLinkAction) {
+                let arguments = absoluteURLString.components(separatedBy: EventFormatterLinkActionSeparator)
```

Comment:
Don't you think we should force "shouldDoAction = false" here too?
(I checked the original code in the `RoomViewController`,  they didn't force this flag, but I think this is a mistake )